adsingestp/parsers/copernicus.py

```diff
@@ -0,0 +1,209 @@
+    def _parse_ids(self):
+        self.base_metadata["ids"] = {}
+
+        issns = []
+        if self.input_metadata.find("issn"):
+            issns.append(("print", self.input_metadata.find("issn").get_text()))
+
+        if self.input_metadata.find("eissn"):
+            issns.append(("electronic", self.input_metadata.find("issn").get_text()))
+
+        self.base_metadata["issn"] = issns
+
+        if self.input_metadata.find("doi"):
+            self.base_metadata["ids"]["doi"] = self.input_metadata.find("doi").get_text()
+
+    def _parse_title(self):
+        title_array = self.input_metadata.find_all("article_title")
+        if title_array:
+            title_array_text = [i.get_text() for i in title_array]
+            # use BS to remove html markup inside title field
+            if len(title_array) == 1:
+                title_temp = BeautifulSoup(title_array_text[0], "html.parser")
+                title = title_temp.get_text().title()
+
+                self.base_metadata["title"] = title
+            else:
+                title_temp = BeautifulSoup(": ".join(title_array_text), "html.parser")
+                title = title_temp.get_text().title()
+
+                self.base_metadata["title"] = title
+
+        else:
+            raise MissingTitleException("No title found")
+
+    def _parse_author(self):
+        author_list = []
+        name_parser = utils.AuthorNames()
+
+        affil_map = {}
+
+        # Create a dictionary to map affiliation names to affiliation numbers
+        if self.input_metadata.find_all("affiliations"):
+            affil_list = self.input_metadata.find_all("affiliations")[0].find_all("affiliation")
+            for aff in affil_list:
+                affil_map[aff.get("numeration", "")] = self._clean_output(aff.get_text())
+
+        author_array = self.input_metadata.find_all("author")
+        for a in author_array:
+            author_temp = {}
+            name = a.find("name").get_text()
+            parsed_name = name_parser.parse(
+                name, collaborations_params=self.author_collaborations_params
+            )
+            author_temp = parsed_name[0]
+
+            if a.find("email"):
+                author_temp["email"] = a.find("email").get_text()
+            if a.find("contrib-id"):
+                author_temp["orcid"] = a.find("contrib-id").get_text()
+
+            if a["affiliations"]:
+                aff_temp = []
+                for author_affil in str(a["affiliations"]).split(","):
+                    aff_temp.append(affil_map[author_affil])
+                author_temp["aff"] = aff_temp
+
+            author_list.append(author_temp)
+
+        if not author_list:
+            raise MissingAuthorsException("No contributors found for")
+
+        self.base_metadata["authors"] = author_list
+
+    def _parse_pubdate(self):
+        if self.input_metadata.find("publication_date"):
+            self.base_metadata["pubdate_electronic"] = self.input_metadata.find(
```

Comment:
This one only does electronic publication?

Reply:
good catch! Updated to reflect the same date for `printDate` as well